---
title: "Exploratory Data Analysis"
subtitle: "Metagenomic Analysis of Fecal Samples as a Differential Diagnostic Tool for Colorectal Cancer"
format: 
  pdf:
    header-includes:
      - \setkomafont{title}{\Large\bfseries}
      - \setkomafont{author}{\normalsize}
      - \setkomafont{date}{\small}
author:
    - name: Kaitlyn Schisler
    - name: Cory Spern
    - name: Faith Shipman
    - name: Jacob Anderson
---

```{r, echo=FALSE}
#| label: Setup
source("Qualifying_Studies_List.R")
suppressPackageStartupMessages({
  library(tidyverse)
  library(knitr)
  library(kableExtra)
  library(patchwork)

  library(vegan)
  library(mia)
  library(scater)
  library(lefser)

  library(grid)
  library(gridExtra)
})

progression_order <- c(
  "HC",
  "PA",
  "PA+",
  "PA-M",
  "CRC",
  "CRC+",
  "CRC-H",
  "CRC-M", 
  "Other"
)

progression_colors <- c(
  "HC" = "#539deb",
  "PA" = "#e4e85b",
  "PA+" = "#e8a25b",
  "CRC" = "#cf1919",
  "CRC+" = "#8f0000",
  "CRC-M" = "#4d0404",
  "CRC-H" = "#520f76",
  "Other" = "#aaa3a3"
)

qualifying_studies <- qualifying_studies %>% 
  mutate(
    disease_class = factor(disease_class, levels = progression_order)
  )
```

### Introduction

The standardization of data collection and processing is the foundation of a strong machine learning algorithm (P. Li et al., 2025). As of December 2025, Manghi et al. (2025) released version 3, their newest version, of their curatedMetagenomicData set (cMD3), which is a manually curated metagenomic data set from 94 studies in 42 countries, totaling 22,710 samples. This was standardized and made available through Bioconductor in R, well known for its package use in genomic and metagenomic studies (Drnevich et al., 2025), and from which previous versions of the cMD have been used in similar studies (Pasolli et al., 2017). Since much of the processing is completed for taxonomic and pathway identification through HUMAnN3 and MetapPhlAn via bioBakery3, it is an ideal data source for jumpstarting our analysis. With there being precedence for using the cMD for this type of work, it remains a fine choice for  This data set, however, includes various sample sites, diseases, and variables irrelevant to our study. Thus, the following inclusionary criteria were applied to the full cMD3 data set using RStudio to determine which studies were to be included in our final subset: (i) sample site is “stool”, (ii) age is ≥18 years, (iii) the disease was not blank, and (iv) the disease contained some text regarding CRC, PA, or (v) the study was HMP_2012, which is an important study for healthy controls from the Human Microbiome Project.

Our goal is to gather retrospective, multicohort study of metagenomic human gut microbiome data to generate a supervised machine learning (ML) model for the detection of CRC progression. This machine learning model will act as a proof-of-concept precursor to a diagnostic tool that can ease the burden of colonoscopy suites for preventative screening, giving a way for doctors to triage patients for effective backlog reduction. It will also guide future research on the influence of microbiomes on disease states.

For our exploratory data analysis, we plan to focus on the following:

1.  **Data availability** - cMD3 returns a TreeSummarizedExperiment object with relative_abundance, gene_families, marker_abundance and presence, and pathway_abundance and coverage data. How much data is available from each study, and how does that affect our ability to perform our analyses and modeling?

2.  **Basic metadata analysis** - All included samples contain information regarding age and gender, which is important considering they are some of the factors in CRC screening and preventative care (Cancer Over Time, n.d.; Duan et al., 2022; Siegel et al., 2025).  Body mass index (BMI) and age are also known to influence each other, thus it was decided to check if there would be any clustering or relationships. Since BMI has been previously associated with microbial shifts (Li et al., 2025), it should be heavily considered when modelling. Geographical location and consequently diet are also well established to be linked to CRC development, and it should be noted that the three most affected countries by CRC are Japan, the United States, and China (Colorectal Cancer Statistics, n.d.; Worldwide Cancer Data, n.d.; Dakal et al., 2025; T. Li et al., 2025; Singh et al., 2017). The following analyses were conducted:

    1.  Normality of age and BMI are analyzed with summary statistics, histograms, and box plots. Analysis of Variance (ANOVA) is used to compare the mean across CRC progression classifiers. These will be performed at a 95% confidence level, and with a large sample size, the central limit theorem allows for an ANOVA to be performed when there is slight skew. When significance arises, Tukey’s Honest Significant Difference (HSD) test is used to determine which pairings are significantly different from each other.

    2.  Gender proportions are compared across age by decade via a faceted stacked bar chart, with a binomial test to determine if the proportion of gender is significantly different in our sample.

    3.  Linear regression models were conducted for each CRC progression classifier against age and BMI to check for potential relationships, which are also represented by scatter plots. These are age-centered to place the results into biological relevance and compared to “HC”.

3.  **Taxonomic and pathway analysis** - using a combination of PERMANOVA and Linear Discriminant Analyses (LDA) for initial identification of any taxonomic or pathway trends in the data.

4.  **Data Limitations** - For investigation into the data limitations that will be encountered during next steps in our model creation allowing further narrowing of scope.

Our Exploratory Data Analysis provides the foundation for understanding the cMD3 dataset and our filtered data. Through our EDA we can assess the quality of the data and key characteristics to refine the scope of our project and future modeling.

### Exploratory Data Analysis

#### Data Availability

Table 1 summarizes the data available across the CRC-relevant studies included in this analysis after filtering. Each study contains between 59 and 616 samples, with the largest cohort being YachidaS_2019 and the smallest being ThomasAM_2018b. The studies span multiple geographic regions across Asia, Europe and North America, providing a diverse representation of populations relevant to colorectal cancer research.

All studies contain taxonomic relative abundance profiles, with between 292 and 740 taxa per study, allowing for cross-study comparisons of microbial composition. Functional metagenomic information is also available in the form of gene family, marker gene, and pathway abundance tables, although these datasets vary substantially in size with each value in these columns representing the available amount of features per study, reflecting the high dimensionality of functional genomic profiling from shotgun metagenomic sequencing. This high dimensionality also requires additional exploration and filtering before modelling, and increases computational demands. Variation in geographic representation also introduces heterogeneity that must also be considered in later analyses.

#### Metadata Analysis

The age summary statistics in Appendix 1-A show a mean of 59, standard deviation of 14.75, and median of 63, suggesting a skew. When faceted by gender, aside from a binomial test showing statistical difference in proportion (41.1% female, p = 5.821e-14, Appendix 1-B and 3-A), the median, mean, and quartiles are not remarkably different. A histogram and box plot (Figure 1-A and 1-B) were generated to corroborate this information (Q-Q plot available, Appendix 2-A). While age has a slight left-tailed skew with a bimodal shape, seemingly caused by the healthy controls visible in the box plot, it is not to an extreme that would harm our final model detrimentally. Skew and variance of age was checked between the sexes via a faceted histogram, which can be found in Appendix 2-B. An ANOVA found a statistical difference in the mean ages of the groups complementary to the box plot (95% CI, p = \<2e-16, Appendix 3-B), requiring a follow-up Tukey’s HSD test to determine which pairings were significantly different from each other. The following pairings with "HC" were found to be statistically different: "Other", "PA", "PA+", "CRC", and "CRC+" (all p-values \<0.000, Appendix 3-B). All other differences in age between the CRC progression classifiers showed no statistical significance at a 95% confidence interval.

After checking normality of the data, the age was divided into decades, with those 18 and 19 years old included in the 20-29 group to avoid a decade being visualized with only two years represented, and all those 80 years or older being grouped together as there was only one individual above 89 years. The two least represented decades in this study will be the 80+ and 30-39 decades (2.7% and 4.0%, respectively) and the three most represented being 60-69, 70-79, and 50-59 (34.4%, 21.8%, 19.9%, respectively; Appendix 1-C). Figure 1-C represents this data in proportions for easier comparisons across the decades, faceted by gender, as it is well established that males have a higher relative risk of CRC than females (Colorectal Cancer Statistics, n.d.; Duan et al., 2022). This pattern is reflected across most age groups in our data, particularly from 30-79. Furthermore, visually there are less samples with recorded comorbidities, as well as less "PA" compared to "CRC" and "HC", overall.

```{r, echo=FALSE}
#| label: Age Normality & CRC Variation across Age/Gender
#| fig-width: 14
#| fig-height: 7
#| fig-dpi: 500
# ---

# NORMALITY ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#########################################

# Histogram
###########
normality1 <- qualifying_studies %>%
  # Set Up
  # ------
filter(!is.na(age)) %>%
  # Plotting
  # --------
ggplot(aes(x = age)) + 
  geom_histogram(bins=20) + 
  # Labeling
  #---------
labs(title = "Distribution of Age: Overall", x = "Age", y = "Count") +
  theme(plot.title = element_text(hjust = 0, size = 10), 
        axis.title = element_text(size = 10))

# Box Plot: Normality by CRC Category
#####################################
normality2 <- qualifying_studies %>%   
  # Set Up   
  # ------   
  filter(!is.na(age)) %>%   
  # Plotting   
  # --------   
  ggplot(aes(x=disease_class, y=age)) +    
  geom_boxplot(outlier.shape = NA) +   
  geom_jitter(alpha = 0.07, width = 0.15) +   
  stat_summary(fun = mean, geom = "point", shape = 20, size = 5, color = "red") +
  # Labeling   
  #--------- 
  labs(title = "Distribution of Age: By CRC Progression Classifier", 
       x = "CRC Progression Classifier", y = "Age") +
  theme(plot.title = element_text(hjust = 0, size = 10), 
        axis.title = element_text(size = 10))

# CRC ACROSS AGE/GENDER ~~~~~~~~~~~~~~~~~
#########################################

# Gendered Barplot: Proportions
###############################
decade1 <- qualifying_studies %>%
  # Set Up
  # ------
filter(!is.na(age_decade), !is.na(gender)) %>%
  # Plotting
  # --------
ggplot(aes(x = age_decade, fill = disease_class)) + 
  geom_bar(position = "fill") +
  scale_fill_manual(values = progression_colors, 
                    name = "CRC Progression \nClassifiers") +
  facet_wrap(~gender) +
  # Labeling
  #---------
labs(title = "CRC Progression Proportion by Age Decade, Grouped by Gender", 
     x = "Age by Decade", y = "Proportion") +
  theme(plot.title = element_text(size = 12))

###################
# Display Patchwork
###################
layout1 <- c( # t, l , b = t, r = l
  area(1, 1, 2, 2), 
  area(1, 3, 2, 4),
  area(3, 1, 5, 4), 
  area(3, 5, 5, 5)
)

#plot(layout1)

age_gender_caption <- str_wrap(
  "(A) Histogram demonstrating a bimodal age distribution with a secondary peak near 25 years and a primary peak near 60 years. 
   (B) Box plot of age across CRC progression classifiers, with jittered transluscent points to demonstrate spread of the sample. The red dot represents the mean, the line represents the median.
   (C) Bar plot displaying the gender faceted proportion of CRC progression classifiers across each decade. Notable differences include and visual increase in HC from 18-69 for females, and increased PA and CRC/CRC+ across most decades in males.", 
  width = 150
)

normality1 + normality2 + decade1 + guide_area() + 
  plot_layout(design = layout1, guides = "collect", widths = c(1,1,1,1,0.7)) +
  plot_annotation(
    title = "Figure 1: Age Normality & CRC Variation across Age and Gender",
    tag_levels = "A",
    caption = age_gender_caption
  ) &
  theme(
        plot.title = element_text(hjust = 0, size = 14),
        plot.caption = element_text(hjust = 0, size = 12), 
        legend.key.size = unit(0.4, "cm"), 
        legend.title = element_text(size = 12)
  )
```

The three most affected countries by CRC are well represented in this multi-cohort study (Appendix 1-D and 2-E), and within these three target countries, there is representation from 6 of our classifiers: "Other", "HC", "PA", "CRC", "CRC+", and "CRC-H". The only unrepresented classifier is "PA+".

Figure 2-A shows the distribution of BMI among the CRC progression classifiers; while there were a few outliers, particularly in the upper BMI bounds of the "PA", "PA+", and "CRC", there are visually striking differences among the means. An ANOVA (Appendix 3-C) showed statistical significance (95% confidence, p = \<2e-16), warranting a Tukey's HSD for further investigation. Of 21 pairings, 13 were found to be statistically significant to a 95% confidence level, meaning BMI could have an impact. It appears to have the most impact on "CRC-H" (5 pairings), "Other" (4 pairings), and "CRC" (4 pairings). BMI and age were also plotted against one another, shown in Figure 2-B with the associated linear models for each CRC progression classifier represented on the scatter plots (Appendix 3-D). Results of the age-centered, linear regression modelling shows a very low R-squared value of 0.0707, meaning that BMI and age only explains about 7% of the variance, though there are a few relationships that show statistical significance to a 95% confidence level, such as in "CRC-H" and "Other". With such a low explanation of the variance, it would be interesting to see how this may or may not extend towards the taxa and pathway relationships in the gut microbiome.

```{r, echo=FALSE}
#| label: BMI Associations
#| fig-width: 14
#| fig-height: 7
#| fig-dpi: 500
# ---

# BMI ~~~~~~~~~~~~~~~~~~~~~~
############################

# Box Plot 
##########
bmi1 <- qualifying_studies %>%   
  # Set Up   
  # ------   
  filter(!is.na(BMI)) %>%   
  # Plotting   
  # --------   
  ggplot(aes(x=disease_class, y=BMI)) +    
  geom_boxplot(outlier.shape = NA) +   
  geom_jitter(alpha = 0.15, width = 0.15) +   
  stat_summary(fun = mean, geom = "point", shape = 20, size = 5, color = "red") +
  # Labeling   
  #--------- 
  labs(
    title = "Distribution of BMI by CRC Progression Classifier", 
    x = "CRC Progression Classifiers", 
    y = "BMI"
  ) +
  theme(plot.title = element_text(size = 12))
    

# Scatterplot by CRC Progression 
################################
bmi2 <- qualifying_studies %>% 
  # Set Up   
  # ------   
  filter (!is.na(age), !is.na(BMI)) %>% 
  # Plotting   
  # --------   
  ggplot(aes(x=age, y=BMI)) +    
  geom_point(aes(color = disease_class, alpha = 0.7)) + 
  geom_smooth(method = "lm", se = TRUE, formula = y ~ x, color = "grey40") +
  scale_color_manual(values = progression_colors, 
                     name = "CRC Progression Classifiers") +
  facet_wrap(~disease_class) +
  # Labeling   
  #---------  
  labs(
    title = "Age by BMI, Colored by CRC Progression Classifier", 
    x = "Age", 
    y = "BMI"
  ) +
  theme(plot.title = element_text(size = 12), 
        legend.position = "none")

# Display
#########
layout2 <- c( # t, l , b = t, r = l
  area(1, 1, 2, 3), 
  area(1, 4, 2, 6)
)

#plot(layout2)

BMI_caption <- str_wrap(
  "(A) Box plot of BMI by CRC progression classifier. The red dot marks the mean, while the black line marks the median. All classifiers generally have a normal distribution.
   (B) Scatter plots of age by BMI, with a linear model regression line of y ~ x, faceted by CRC progression classifier. Notice that there is minimally notable clustering.", 
  width = 200
)

bmi1 + bmi2 +
  plot_layout(design = layout2) +
  plot_annotation(
    title = "Figure 2: BMI Associations with CRC Progression Classifiers",
    tag_levels = "A",
    caption = BMI_caption
  ) &
  theme(
        plot.title = element_text(hjust = 0, size = 14),
        plot.caption = element_text(hjust = 0, size = 12), 
        legend.key.size = unit(0.5, "cm"), 
        legend.title = element_text(size = 10), 
        legend.direction = "horizontal"
  )
```

#### Taxonomic & Pathway Analysis

The combined filtered data sets from cMD3 contain over 1000 taxa. To identify any trends in the data, we looked at beta-diversity, which is a test for dissimilarity between the study conditions. Although the p-value was highly significant (P = 0.001), the R-squared value (R\^2 = 0.0078) indicates much of the dissimilarity is not explained by the study conditions.

```{r}
#| label: PCoA Plot
#| echo: false
#| message: false
#| warning: false
#| fig-align: center
#| fig-height: 6
#| fig-width: 14

data <- returnSamples(
  qualifying_studies,
  dataType = "relative_abundance",
  counts = TRUE
)

# Calculate distance matrix
dist_matrix <- vegdist(t(assay(data, "relative_abundance")), 
                       method = "bray")

# Get metadata
metadata <- colData(data) %>% as.data.frame()

# PERMANOVA test
permanova_result <- adonis2(dist_matrix ~ study_condition, 
                            data = metadata, 
                            permutations = 999)

# Extract R-squared and p-value
r_squared <- permanova_result$R2[1]
p_value <- permanova_result$`Pr(>F)`[1]

PCoA_caption <- str_wrap(
  "Principle Coordinate Analysis illustrating Bray-Curtis dissimilarity of metagenomic data from cMD3 study. Comparisons were made between study conditions and age groups.  ", 
  width = 200
)


data %>% 
  agglomerateByRanks() %>% 
  runMDS(FUN = vegdist, method = "bray", exprs_values = "relative_abundance", altexp = "genus", name = "BrayCurtis") %>% 
  plotReducedDim("BrayCurtis", colour_by = "study_condition", shape_by = "age_decade") + 
  labs(x = "PCo 1", y = "PCo 2", subtitle = glue::glue("PERMANOVA: R² = {round(r_squared, 4)}, p = {round(p_value, 4)}"),
       caption = PCoA_caption) + 
  guides(colour = guide_legend(title = "Study Condition"), shape = guide_legend(title = "Age (by decade)")) + 
  plot_annotation(title = "Figure 3: Beta Diversity of Samples by Age and Study Condition") +
  theme(legend.position = c(0.01, 0.28), 
        axis.text = element_text(size = 12), 
        axis.title = element_text(size = 14), 
        legend.text = element_text(size = 12), 
        legend.title = element_text(size = 13), 
        plot.subtitle = element_text(size = 12),
        plot.caption = element_text(hjust = 0))


```

To understand how the taxa are distributed across the different study conditions, two relative abundance plots were generated at the family and genus level.

```{r}
#| label: Taxon by Condition, Genus and Family
#| echo: false
#| warning: false
#| message: false
#| fig-width: 14
#| fig-height: 6
#| fig-align: center

data <- returnSamples(
  qualifying_studies,
  dataType = "relative_abundance",
  counts = TRUE
)

fareblind15 <- c("#000000","#004949","#009292","#FF6DB6","#FFB6DB",
                   "#490092","#006DDB","#B66DFF","#6DB6FF","#B6DBFF",
                   "#920000","#924900","#DB6D00","#24FF24","#FFFF6D") 

# Extract and aggregate
abund_data <- assay(data, "relative_abundance")
sample_meta <- colData(data) %>% as.data.frame()
tax_data <- rowData(data) %>% as.data.frame()

# Convert to long format
abund_long <- abund_data %>%
  as.data.frame() %>%
  rownames_to_column("taxon") %>%
  pivot_longer(-taxon, names_to = "sample", values_to = "abundance")

# Add metadata
sample_meta$sample <- rownames(sample_meta)

# Merge and aggregate
abund_agg <- abund_long %>%
  merge(sample_meta[, c("sample", "study_condition")], by = "sample") %>%
  group_by(taxon, study_condition) %>%
  summarise(abundance = mean(abundance), .groups = "drop")

#######################################################
#Group by
# Extract genus level
abund_long_genus <- abund_data %>%
  as.data.frame() %>%
  rownames_to_column("taxon") %>%
  mutate(genus = str_extract(taxon, "g__[^|]*")) %>%  # Extract genus
  mutate(genus = str_remove(genus, "g__")) %>%  # Remove the g__ prefix
  pivot_longer(-c(taxon, genus), names_to = "sample", values_to = "abundance")

# Then aggregate by genus instead of taxon
abund_agg <- abund_long_genus %>%
  merge(sample_meta[, c("sample", "study_condition")], by = "sample") %>%
  group_by(genus, study_condition) %>%
  summarise(abundance = sum(abundance), .groups = "drop")  # Sum instead of mean

# Get top 15 genera
top_genera <- abund_long_genus %>%
  group_by(genus) %>%
  summarise(mean_abund = mean(abundance)) %>%
  arrange(desc(mean_abund)) %>%
  slice_head(n = 15) %>%
  pull(genus)

# Filter and normalize
abund_normalized <- abund_agg %>%
  filter(genus %in% top_genera) %>%
  group_by(study_condition) %>%
  mutate(relative_abundance = abundance / sum(abundance)) %>%
  ungroup()

# Plot
p1 <- abund_normalized %>%
  ggplot(aes(x = study_condition, y = relative_abundance, fill = genus)) +
  geom_col() +
  scale_fill_manual(values = fareblind15) +
  coord_flip() +
  ggtitle("Gut microbiota by condition (Genus level)") +
  theme_minimal()


######################################
# Group by Family
abund_long_family <- abund_data %>%
  as.data.frame() %>%
  rownames_to_column("taxon") %>%
  mutate(family = str_extract(taxon, "f__[^|]*")) %>%  # Extract family
  mutate(family = str_remove(family, "f__")) %>%  # Remove the g__ prefix
  pivot_longer(-c(taxon, family), names_to = "sample", values_to = "abundance")

# Then aggregate by family instead of taxon
abund_agg <- abund_long_family %>%
  merge(sample_meta[, c("sample", "study_condition")], by = "sample") %>%
  group_by(family, study_condition) %>%
  summarise(abundance = sum(abundance), .groups = "drop")  # Sum instead of mean

# Get top 15 genera
top_genera <- abund_long_family %>%
  group_by(family) %>%
  summarise(mean_abund = mean(abundance)) %>%
  arrange(desc(mean_abund)) %>%
  slice_head(n = 15) %>%
  pull(family)

# Filter and normalize
abund_normalized <- abund_agg %>%
  filter(family %in% top_genera) %>%
  group_by(study_condition) %>%
  mutate(relative_abundance = abundance / sum(abundance)) %>%
  ungroup()

# Plot
p2 <- abund_normalized %>%
  ggplot(aes(x = study_condition, y = relative_abundance, fill = family)) +
  geom_col() +
  scale_fill_manual(values = fareblind15) +
  coord_flip() +
  ggtitle("Gut microbiota by condition (Family level)") +
  theme_minimal()



figure_4_caption <- str_wrap(
  "Stacked bar plots of cMD3 metagenomic data based on study conditions of filtered studies (A) Family taxonomic level
   (B) Genus taxonomic level.", 
  width = 200
)

p2 + p1 +
  plot_annotation(
    title = "Figure 4: Relative Abundance of cMD3 Study Metagenomic Data at Family and Genus taxonomic levels",
    caption = figure_4_caption,
    tag_levels = "A"
  ) +
  theme(plot.caption = element_text(hjust = 0))

```

The relative abundance stacked bar plots also indicate there was very little difference at the family and genus taxonomic levels. Considering these data sets cover several countries and age categories, it is possible that further separation into more groups may provide better comparisons. As Japan and the United States contain the greatest number of participants, these countries were separated out. Then, for the same reason, we compared only the control and CRC groups, since these health conditions also have the highest sample sizes. To identify key differences in how taxa may be enriched between groups, differential abundance was looked at using LDA Effect Size (LEfSe) analysis. Using a reference (control) and a treatment (CRC), LEfSe performs Kruskal-Wallis test to identify taxa that were statistically different from zero between the control and CRCgroups. Taxa found to be statistically significant were run with a Wilconox Rank-Sum Test to perform pairwise comparisons. Finally, the effect size is calculated using LDA.

```{r}
#| label: Taxon by Country
#| echo: false
#| warning: false
#| message: false
#| fig-width: 14
#| fig-height: 6
#| fig-align: center

#Filter data for Japan
data_JPN <- data[, data$country == "JPN"]

data_JPN <- data_JPN[, data_JPN$study_condition == c("control","CRC")]

data_JPN_genus <- agglomerateByRank(data_JPN, rank = "genus")


p1b <- lefser(
  relativeAb(data_JPN_genus),
  kruskal.threshold = 0.05,
  wilcox.threshold = 0.05,
  lda.threshold = 2,
  classCol = "study_condition",
  subclassCol = NULL,
  assay = 1L,
  trim.names = FALSE,
  checkAbundances = TRUE
) %>%
  lefserPlot() +
  labs(title = "Japan - CRC vs Control")

#Filter data for United States
data_USA <- data[, data$country == "USA"]

data_USA <- data_USA[, data_USA$study_condition == c("control","CRC")]

data_USA_genus <- agglomerateByRank(data_USA, rank = "genus")


p2b <- lefser(
  relativeAb(data_USA_genus),
  kruskal.threshold = 0.05,
  wilcox.threshold = 0.05,
  lda.threshold = 2,
  classCol = "study_condition",
  subclassCol = NULL,
  assay = 1L,
  trim.names = FALSE,
  checkAbundances = TRUE
) %>%
  lefserPlot() +
  labs(title = "United States - CRC vs Control")

# Save lefser results before plotting
res_JPN <- lefser(
  relativeAb(data_JPN_genus),
  kruskal.threshold = 0.05,
  wilcox.threshold = 0.05,
  lda.threshold = ,
  classCol = "study_condition"
)

res_USA <- lefser(
  relativeAb(data_USA_genus),
  kruskal.threshold = 0.05,
  wilcox.threshold = 0.05,
  lda.threshold = 2,
  classCol = "study_condition"
)

# Filter for CRC-associated taxa (positive LDA scores, assuming CRC is the positive class)
crc_JPN <- res_JPN[res_JPN$scores > 0, "features"]
crc_USA <- res_USA[res_USA$scores > 0, "features"]

p1b + p2b

# Find common taxa
#intersect(crc_JPN, crc_USA)

#bugsInCommon <- intersect(crc_JPN, crc_USA)

figure_5_caption <- str_wrap(
  "Differential abundance of cMD3 Study metagenomic data using LDA Effect Size (LEfSe) comparing healthy controls and subjects with colorectal cancer (CRC) in (A) Japan
  and the (B) United States.", 
  width = 200
)

p1b + p2b +
  plot_annotation(
    title = "Figure 5: Differential Abundance of cMD3 Study Metagenomic Data at Genus taxonomic level for Japan and the United States",
    caption = figure_5_caption,
    tag_levels = "A"
  ) +
  theme(plot.caption = element_text(hjust = 0))


```

Three genera (Blautia, Collinsella, and Dorea) were found to have greater abundance in CRCsamples compared to controls. This warrants further investigation and may help guide the taxonomic portion of the machine learning model.

#### Data Limitations

While this work assesses the feasibility of leveraging human gut microbiome “fingerprints” for disease-state characterization, limitations do still exist from data availability, metadata completeness, and necessary inclusion criteria. All of which are not unique to this study but rather showcase the current challenges amongst large scale microbiome meta-analyses. 

A rather major limitation is the incomplete availability of clinically and biologically relevant metadata across the various studies in our analysis. The metadata completeness analysis shown below visualises this. There is only a subset of variables that are consistently available throughout the data such as the sequencing platform use, participant country, gender, age and age by decade, BMI, and disease class. This is unfortunate because it is known that the microbiome is drastically influenced by variables that are not consistent such as antibiotic use, smoking status, alcohol consumption, and disease stages (Tegegne & Savidge, 2025). The original use case was intended to be applied to antibiotic usage which will unfortunately need to be pushed until more complete data sets are available. 

While this study focuses on CRC metagenomic analysis/”fingerprinting”, the available data has either empty or missing columns for things like tumor stage, location, and metastasis. Original plans had the separation of CRC patients with and without metastasis which is currently not feasible until more complete data is available. This will result in a computational framework that can only compare broad disease categories rather than detailed cancer stages. 

```{r, echo=FALSE}

#| label: Metadata heat map 


#creating a metadata heatmap showing some variables that aren't used downstream
meta_vars <- c(
  # Demographics
  "age", "age_decade", "gender", "BMI",
  
  # Lifestyle / exposures
  "diet", "alcohol", "alcohol_numeric", "smoker", "ever_smoker",
  "antibiotics_current_use",
  
  # Clinical context
  "disease_class", "disease_stage", "disease_location",
  
  # Geography
  "country", "location", "population",
  
  # Technical
  "sequencing_platform", "DNA_extraction_kit",
  
  # Timing
  "days_from_first_collection", "days_after_onset"
)

# Confirm they exist
#meta_vars[meta_vars %in% colnames(qualifying_studies)]

#compute metadata completeness (percent of non-mising data)

meta_completeness <- qualifying_studies %>%
  summarise(across(all_of(meta_vars), ~ mean(!is.na(.)))) %>%
  pivot_longer(
    cols = everything(),
    names_to = "variable",
    values_to = "completeness"
  ) %>%
  mutate(completeness = completeness * 100)

#print(meta_completeness)

#create a heatmap for metadata completeness 
#library(ggplot2)
meta_completeness <- meta_completeness %>%
  arrange(completeness) %>%
  mutate(variable = factor(variable, levels = variable))

ggplot(meta_completeness,
       aes(x = "", y = variable, fill = completeness)) +
  geom_tile(color = "white") +
  scale_fill_gradient(
    low = "white",
    high = "steelblue",
    limits = c(0, 100),
    name = "% samples\nwith data"
  ) +
  labs(
    title = str_wrap("Metadata Completeness in CRC Related Metagenomic Samples"),
    subtitle = str_wrap("Percent of samples with non-missing values per variable"),
    x = NULL,
    y = NULL
  ) +
  theme_bw() +
  theme(
    axis.text.x = element_blank(),
    axis.ticks.x = element_blank(),
    #center justify titles
    plot.title = element_text(face = "bold", size = 12, hjust=0.5),
    plot.subtitle = element_text(size = 10, hjust=0.5)
  )
```

Other limitations came about after the filtering of the dataset to keep data consistent and comparable. Filters included only adult patients, stool samples only, disease labels, and CRC relevant studies. While the data quality is better using these filters, it does shrink the dataset. The filtering created a working database with mostly CRC samples where polyp-only and adenoma cases are few and far between. Due to this, disease progression from polyp to cancer modeling will not be reliable. Again this leads to a focus on broad comparisons such as healthy participants vs early lesions vs CRC. 

As this is an aggregated data set, there are other considerations to factor in, specifically the difference in laboratory techniques, sequencing machines, extraction kits, and protocols used throughout all of the studies. These variables are documented but they cannot be fully corrected for so they must be mentioned as they create biological noise in an exploratory modeling approach such as this.

```{r, echo=FALSE}
#| label: Sequencing variations


#create ggplot of studies selected and sequencing platform and extraction kit used
#summary table with counts per study X
study_tech_summary <- qualifying_studies %>%
  filter(
    !is.na(sequencing_platform),
    !is.na(DNA_extraction_kit)
  ) %>%
  count(study_name, sequencing_platform, DNA_extraction_kit)
    #sanity check lol
#head(study_tech_summary)
#create ggplot
ggplot(
  study_tech_summary,
  aes(
    y = study_name,
    x = n,
    fill = DNA_extraction_kit
  )
) +
  geom_col() +
  facet_wrap(~ sequencing_platform, scales = "free_y") +
  theme_bw() +
  labs(
    title = str_wrap("Sequencing Platforms and DNA Extraction Kits Across CRC-Relevant Studies", width = 50),
    x = "Number of samples",
    y = "Study name",
    fill = "DNA extraction kit"
  ) +
  theme(
    # Centers the main title
    plot.title = element_text(hjust = 0.5, face = "bold")
  )
```

Proposal of this analytical framework is intentionally framed as a proof-of-concept tool rather than a ready to use clinically diagnostic model to show the idea works in principle. Biggest limitations currently include missing or inconsistent metadata, not from the modeling itself. This is to say that when future datasets with more consistent data become available the framework can handle expansions.

### Conclusion

Exploration that occurred during this phase of our study revealed that after filtering for our inclusionary criteria, the 12 studies that would remain in our multi-cohort retrospective study contains taxonomic and pathway data required for modelling purposes. We successfully audited the taxonomic and metadata, revealing both anticipated and surprising patterns. While biometric and demographic data such as age, BMI, and gender appear to have some influence over the CRC progression classifiers, the predictive value is not as substantial as we may have assumed given prior research. Diet has also been shown to influence the human microbiome, however it is not often collected as a data point in studies, which is reflected in this data set, with only 276 data points containing information regarding diet.  Approximately 98.4% classified as omnivores and the remainder (10 samples) representing vegetarians, which may not be enough data points to investigate variability in diet as a factor in CRC progression. This may over inflate proportions of different stages of the disease, like in various visualizations attempted (Appendix 2-C).

These initial analyses also revealed several imbalances. For example, countries are highly represented in our data (Figure 6-B), though 38.9% of our study belongs to Japan. The three most affected countries by CRC are well represented in this multi-cohort study (Figure 2-A, Appendix 1-D), which is important to ensure that our results and model will be applicable to the most vulnerable target populations. However, three continents are missing: Africa, Oceania, and South America. This will need to be considered in future evaluations of the model, with particular attention when searching for data to include in training sets.

Proposal of this analytical framework is intentionally framed as a proof-of-concept tool rather than a ready to use clinical diagnostic model to show the idea works in principle. Biggest limitations currently include missing or inconsistent metadata, not from the modeling itself. This is to say that when future datasets with more consistent data become available during further version releases of the cMD or more additions via the pipelines and framework provided by Manghi et al. (2025) the proof-of-concept tool can handle expansions.

Due to the bimodality in age caused by the healthy controls, if there is no real difference in microbial fingerprints in the healthy controls when the inclusion criteria is changed from 18+ to 30+, proposed computational modeling may benefit from selection of only participants 30+.  In conjunction with various limitations in the selected data which also became apparent leading to the need to pivot with select variables such as exclusion of diet focus, disease stage, antibiotic use, and CRC with individual comorbidities analyses. Future analysis will focus on using Random Forest models as they are robust to this to unbalanced data (Bradter et al., 2022). 

Unfortunately, due to the computing power of our personal computers, we were unable to dig into the pathway data as initial implementation failed. RAM requirements will necessitate a high performance computing cluster like that of Sol at Arizona State University (Jennewein et al., 2023), and thus gaining joint access as a team will be a high priority for completing this analysis and for further modelling needs.

### References

Bradter, U., Altringham, J. D., Kunin, W. E., Thom, T. J., O’Connell, J., & Benton, T. G. (2022). Variable ranking and selection with random forest for unbalanced data. Environmental Data Science, 1, e30. https://doi.org/10.1017/eds.2022.34 

Colorectal cancer statistics. (n.d.). World Cancer Research Fund. Retrieved January 31, 2026, from https://www.wcrf.org/preventing-cancer/cancer-statistics/colorectal-cancer-statistics/ 

Dakal, T. C., Xu, C., & Kumar, A. (2025). Advanced computational tools, artificial intelligence and machine-learning approaches in gut microbiota and biomarker identification. Frontiers in Medical Technology, 6, 1434799. https://doi.org/10.3389/fmedt.2024.1434799 

Drnevich, J., Tan, F. J., Almeida-Silva, F., Castelo, R., Culhane, A. C., Davis, S., Doyle, M. A., Geistlinger, L., Ghazi, A. R., Holmes, S., Lahti, L., Mahmoud, A., Nishida, K., Ramos, M., Rue-Albrecht, K., Shih, D. J. H., Gatto, L., & Soneson, C. (2025). Learning and teaching biological data science in the Bioconductor community. PLOS Computational Biology, 21(4), e1012925. https://doi.org/10.1371/journal.pcbi.1012925 

Jennewein, D. M., Lee, J., Kurtz, C., Dizon, W., Shaeffer, I., Chapman, A., Chiquete, A., Burks, J., Carlson, A., Mason, N., Kobawala, A., Jagadeesan, T., Basani, P. B., Battelle, T., Belshe, R., McCaffrey, D., Brazil, M., Inumella, C., Kuznia, K., … Yalim, J. (2023). The Sol Supercomputer at Arizona State University. Practice and Experience in Advanced Research Computing, 296–301. https://doi.org/10.1145/3569951.3597573 

Li, P., Li, M., & Chen, W.-H. (2025). Best practices for developing microbiome-based disease diagnostic classifiers through machine learning. Gut Microbes, 17(1), 2489074. https://doi.org/10.1080/19490976.2025.2489074 

Li, T., Coker, O. O., Sun, Y., Li, S., Liu, C., Lin, Y., Wong, S. H., Miao, Y., Sung, J. J. Y., & Yu, J. (2025). Multi-Cohort Analysis Reveals Altered Archaea in Colorectal Cancer Fecal Samples Across Populations. Gastroenterology, 168(3), 525-538.e2. https://doi.org/10.1053/j.gastro.2024.10.023 

Manghi, P., Antonello, G., Schiffer, L., Golzato, D., Wokaty, A., Beghini, F., Mirzayi, C., Long, K., Gravel-Pucillo, K., Piccinno, G., Gamboa-Tuz, S. D., Bonetti, A., D’Amato, G., Azhar, R., Eckenrode, K., Zohra, F., Giunchiglia, V., Keller, M., Pedrotti, A., … Waldron, L. (2025). Meta-analysis of 22,710 human microbiome metagenomes defines an oral-to-gut microbial enrichment score and associations with host health and disease. Nature Communications, 17(1), 196. https://doi.org/10.1038/s41467-025-66888-1 

Pasolli, E., Schiffer, L., Manghi, P., Renson, A., Obenchain, V., Truong, D. T., Beghini, F., Malik, F., Ramos, M., Dowd, J. B., Huttenhower, C., Morgan, M., Segata, N., & Waldron, L. (2017). Accessible, curated metagenomic data through ExperimentHub. Nature Methods, 14(11), 1023–1024. https://doi.org/10.1038/nmeth.4468 

Singh, R. K., Chang, H.-W., Yan, D., Lee, K. M., Ucmak, D., Wong, K., Abrouk, M., Farahnik, B., Nakamura, M., Zhu, T. H., Bhutani, T., & Liao, W. (2017). Influence of diet on the gut microbiome and implications for human health. Journal of Translational Medicine, 15(1), 73. https://doi.org/10.1186/s12967-017-1175-y 

Tegegne, H. A., & Savidge, T. C. (2025). Leveraging human microbiomes for disease prediction and treatment. Trends in Pharmacological Sciences, 46(1), 32–44. https://doi.org/10.1016/j.tips.2024.11.007 

Worldwide cancer data. (n.d.). World Cancer Research Fund. Retrieved January 31, 2026, from https://www.wcrf.org/preventing-cancer/cancer-statistics/worldwide-cancer-data/
